routes/index.js

```diff
@@ -38,22 +45,121 @@ router.get('/login', function (req, res, next) {
+        res.render('menus/breakfast', {title: 'Breakfast Menu', breakfast: docs});
+    });
+});
+
+router.get('/add-to-cart/breakfast/:id', function (req, res, next) {
+    let bfItemId = req.params.id;
+    let cart = new Cart(req.session.cart ? req.session.cart : {});
+    Breakfast.findById(bfItemId, function (err, bfItem) {
+        if(err){
+            return res.redirect('/', {title: 'Something went wrong.'});
+        }
+        cart.add(bfItem, bfItem.id);
+        req.session.cart = cart;
+        console.log(req.session.cart);
+        res.redirect('/breakfast');
     });
 });
 
 //get lunch page
 router.get('/lunch', function (req, res, next) {
     Lunch.find(function (err, docs) {
-        res.render('lunch', {title:'Lunch Menu', lunch: docs});
+        res.render('menus/lunch', {title:'Lunch Menu', lunch: docs});
+    });
+});
+
+router.get('/add-to-cart/lunch/:id', function (req, res, next) {
+    let lunchItemId = req.params.id;
+    let cart = new Cart(req.session.cart ? req.session.cart : {});
+    Lunch.findById(lunchItemId, function (err, lunchItem) {
+        if(err){
+            return res.redirect('/', {title: 'Something went wrong.'});
+        }
+        cart.add(lunchItem, lunchItem.id);
+        req.session.cart = cart;
+        console.log(req.session.cart);
+        res.redirect('/lunch');
     });
 });
 
 //get dinner menu
 router.get('/dinner', function (req, res, next) {
     Dinner.find(function (err, docs) {
-        res.render('dinner', {title:'Dinner Menu', dinner:docs});
+        res.render('menus/dinner', {title:'Dinner Menu', dinner:docs});
     });
 });
 
+router.get('/add-to-cart/dinner/:id', function (req, res, next) {
+    let dinnerItemId = req.params.id;
+    let cart = new Cart(req.session.cart ? req.session.cart : {});
+    Dinner.findById(dinnerItemId, function (err, dinnerItem) {
+        if(err){
+            return res.redirect('/', {title: 'Something went wrong.'});
+        }
+        cart.add(dinnerItem, dinnerItem.id);
+        req.session.cart = cart;
+        console.log(req.session.cart);
+        res.redirect('/dinner');
+    });
+});
+
+router.get('/shopping-cart', function (req, res, next) {
+    if(!req.session.cart){
+        return res.render('shopping-cart',{foodItems: null});
+    }
+    let cart = new Cart(req.session.cart);
+    res.render('shopping-cart', {foodItems: cart.generateArray(), totalPrice: cart.totalPrice})
+});
+
+router.get('/checkout', isLoggedIn, function (req, res, next) {
+    if(!req.session.cart){
+        return res.redirect('/shopping-cart');
+    }
+    let cart = new Cart(req.session.cart);
+    res.render('checkout',{total: cart.totalPrice});
+});
+
+router.post('/checkout', isLoggedIn, function (req, res, next) {
+    if(!req.session.cart){
+        return res.redirect('/shopping-cart');
+    }
+    let cart = new Cart(req.session.cart);
+    let stripe = require("stripe")("sk_test_Nm4ul2p0g79wCgLhUB5xBZoD");
```

Comment:
Requirements: Exposed Token. Reset it and hide it in an env.

Reply:
@piazzaana Should be a separate issue. This is actually a back end usage of stripe, which is good, but still needs an env var. Please create another issue.

Reply:
@reactivepixel fixed it https://github.com/piazzaana/chicken-fries/pull/66/commits/8db1d74c4b7ce4b1f6c64e4162a2b6566f3f6c51